readme.md

```diff
@@ -270,6 +270,24 @@ Each top-level component receives a `url` property with the following API:
 
 The second `as` parameter for `push` and `replace` is an optional _decoration_ of the URL. Useful if you configured custom routes on the server.
 
+##### With URL object
+The component `<Link>` can also receive an URL object and it will automatically format it to create the URL string.
+
+```jsx
+// pages/index.js
+import Link from 'next/link'
+export default () => (
+  <div>Click <Link href={{ pathname: 'about', query: { name: 'Zeit' }}}<a>here</a></Link> to read more</div>
+)
+```
+
+That will generate the URL string `/about?name=Zeit`, you can use the following parameters:
```

Comment:
How about linking to the existing doc you mentioned (https://nodejs.org/api/url.html#url_url_strings_and_url_objects) instead?

Are all the properties supported? What happens if user passes `auth`, `host`, etc?